src/java/org/apache/cassandra/io/sstable/format/trieindex/TrieIndexSSTableReader.java

```diff
@@ -958,6 +964,7 @@ public static TrieIndexSSTableReader open(Descriptor descriptor, Set<Component>
 
         ValidationMetadata validationMetadata = (ValidationMetadata) sstableMetadata.get(MetadataType.VALIDATION);
         StatsMetadata statsMetadata = (StatsMetadata) sstableMetadata.get(MetadataType.STATS);
+        CompactionMetadata compactionMetadata = (CompactionMetadata) sstableMetadata.get(MetadataType.COMPACTION);
```

Comment:
`types = EnumSet.of(MetadataType.VALIDATION, MetadataType.STATS, MetadataType.HEADER);` doesn't include `COMPACTION`. this will be `null` for sure. If it's intended to skip loading, we can use `null` directly